app/src/main/java/com/postpc/tenq/ui/activities/SongSearchActivity.java

```diff
@@ -60,12 +67,18 @@ public boolean onQueryTextChange(String newText) {
     }
 
     private void setSearchSongRecycler() {
-        SongSearchAdapter adapter = new SongSearchAdapter(track -> model.addTrackToRoomPlaylist(null, track), binding.progressLoadingMore); //TODO pass room
+        SongSearchAdapter adapter = new SongSearchAdapter(track -> model.addTrackToRoomPlaylist(getAuthService().getCurrentUser(), room, track), binding.progressLoadingMore);
         model.getResults().observe(this, adapter::submitList);
 
         binding.recyclerSongs.setLayoutManager(new LinearLayoutManager(this, RecyclerView.VERTICAL, false));
         binding.recyclerSongs.addOnScrollListener(new PagingScrollListener(model::loadNextPage, binding.progressLoadingMore));
         binding.recyclerSongs.setAdapter(adapter);
     }
 
+    @Override
+    public void onBackPressed() {
+        startActivity(new Intent(SongSearchActivity.this, RoomActivity.class).putExtra("room", room));
```

Comment:
This could be an issue, you're starting an activity from the activity that started this activity... potentially the user can create infinite amount of instances of "RoomActivity" that way. On back press, use "setIntent" (not sure that's the syntax) and then "finish()" to go back to the previous activity